src/charm.py

```diff
@@ -354,10 +478,9 @@ def _handle_ingress_data_provided(self, event: RelationEvent):
 
     def _handle_ingress_data_removed(self, event: RelationEvent):
         """A unit has removed the data we need to provide ingress."""
-        # if this is because the relation is gone, we don't need to do anything
-        if isinstance(event, RelationBrokenEvent):
-            return
-        self._wipe_ingress_for_relation(event.relation)
+        self._wipe_ingress_for_relation(
+            event.relation, wipe_rel_data=not isinstance(event, RelationBrokenEvent)
```

Comment:
Are we sure we want to remove it on a `RelationDepartedEvent` and leave it on a `RelationBrokenEvent`?